### PR TITLE
Document how to start Patroni with an existing data directory

### PR DESCRIPTION
**Pull request by EtienneM**
Following my proposition in [this issue](https://github.com/zalando/patroni/issues/913#issuecomment-452664517), here is a documentation about how to start Patroni with an existing data directory. 
I gathered information from https://github.com/zalando/patroni/issues/526 and https://github.com/zalando/patroni/issues/913 without knowing much about Patroni... 

I am not really sure about where this explanation should go in your documentation, let me know if you want me to move it elsewhere.

Moreover, I don't know how to test locally this documentation so I hope (:crossed_fingers: ) the links are not broken and it shows nicely

### CODE REVIEW

File: docs/existing_data.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+A Patroni cluster can be started with a data directory from a single-node PostgreSQL database. This is achieved by following closely these steps:
+
+#. Manually start PostgreSQL daemon
+#. Create Patroni super-user as defined in the :ref:`authentication <postgresql_settings>` section of the Patroni configuration. If this user is created in SQL, the following queries achieve this:
+
+.. code-block:: sql
+  CREATE USER $PATRONI_SUPERUSER_USERNAME SUPERUSER CREATEDB CREATEROLE REPLICATION BYPASSRLS;
+  ALTER USER $PATRONI_SUPERUSER_USERNAME ENCRYPTED PASSWORD '$PATRONI_SUPERUSER_PASSWORD';
+
+#. Start Patroni (e.g. ``patroni /etc/patroni/patroni.yml``). It automatically detects that PostgreSQL daemon is already running but its configuration is out-of-date.
```
Comment (CyberDem0n) [inline docs/existing_data.rst]: but its configuration is _might be_ out-of-date.

File: docs/existing_data.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+To deploy a Patroni cluster without using a pre-existing PostgreSQL instance, see :ref:`text <link>` instead.
+
+Procedure
+---------
+
+A Patroni cluster can be started with a data directory from a single-node PostgreSQL database. This is achieved by following closely these steps:
+
+#. Manually start PostgreSQL daemon
+#. Create Patroni super-user as defined in the :ref:`authentication <postgresql_settings>` section of the Patroni configuration. If this user is created in SQL, the following queries achieve this:
```
Comment (CyberDem0n) [inline docs/existing_data.rst]: Create Patroni _superuser and replication users_

File: docs/existing_data.rst [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+.. code-block:: sql
+  CREATE USER $PATRONI_SUPERUSER_USERNAME SUPERUSER CREATEDB CREATEROLE REPLICATION BYPASSRLS;
```
Comment (CyberDem0n) [inline docs/existing_data.rst]: CREATE USER $PATRONI_SUPERUSER_USERNAME WITH SUPERUSER ENCRYPTED PASSWORD '$PATRONI_SUPERUSER_PASSWORD';
CREATE USER $PATRONI_REPLICATION_USERNAME WITH REPLICATION ENCRYPTED PASSWORD '$PATRONI_REPLICATION_PASSWORD';

File: docs/existing_data.rst [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+
+FAQ
+---
+
+#. During Patroni startup, Patroni complains that it cannot bind to the PostgreSQL port.
+
+You need to verify ``listen_addresses`` and ``port`` in ``postgresql.conf`` and ``postgresql.listen`` in ``patroni.yml``. Don't forget that ``pg_hba.conf`` should allow such access.
+
+#. After asking Patroni to restart the node, PostgreSQL displays the error message ``could not open configuration file "/etc/postgresql/10/main/pg_hba.conf": No such file or directory``
```
Comment (CyberDem0n) [inline docs/existing_data.rst]: In general, it also might have a meaning that configuration is wrong. It is quite common that postgresql.conf and pg_hba.conf are located in the special config directory (/etc/postgresql/...).
Depending on how you want to manage postgres configuration (with or without Patroni) you can specify either `postgresql.config_dir` to point to the `/etc/postgresql/...` (exact place where postgresql.conf and pg_hba.conf are located) or specify *.conf files in the `postgresql.parameters.hba_file` and `postgresql.custom_conf`. 

You can find a bit more details [here](https://github.com/zalando/patroni/blob/master/docs/SETTINGS.rst#postgresql)

File: docs/existing_data.rst [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+Patroni generates the ``pg_hba.conf`` based on the settings in the :ref:`bootstrap <bootstrap_settings>` section only when it bootstraps a new cluster. In this scenarion the ``PGDATA`` was not empty, therefore no bootstrap happened. This file must exist beforehand.
```
Comment (CyberDem0n) [inline docs/existing_data.rst]: If `postgresql.pg_hba` is specified and `postgresql.parameters.hba_file` is not, then Patroni will owerwrite existing pg_hba.conf in the PGDATA

File: docs/existing_data.rst [hunk 6]
```diff
@@ -0,0 +1,35 @@
+.. _existing_data:
+
+Convert a Standalone to a Patroni Cluster
+=========================================
+
+This section describes the process for converting a standalone PostgreSQL instance into a Patroni cluster.
+
+To deploy a Patroni cluster without using a pre-existing PostgreSQL instance, see :ref:`text <link>` instead.
```
Comment (CyberDem0n) [inline docs/existing_data.rst]:
```
:ref:`text <link>`
```
Not sure where it will bring us :)

**Comment by EtienneM** [comment]
I updated the documentation based on your reviews. I don't really now how to take into account your remarks on the last FAQ qestion

**Pull request merged**